### PR TITLE
Edited PsmFromTsv to ignore enclosing quotes in fragment ion strings

### LINKED ISSUE #2144: MetaDraw can't read .psmtsv's edited in Excel
**Describing the issue**

Tried to open .psmtsv that I sorted in excel in MetaDraw. Excel appends quotation marks to the fragment ions, rendering them unreadable

**Reproducing the error**

Any edit to a .psmtsv in excel reproduces this.

### PR DESCRIPTION
**Pull request by Alexander-Sol**
Closes #2144.

### CODE REVIEW

**Review by avcarr2 — APPROVED** [review]
Looks good.

**Review by nbollis — APPROVED** [review]

**Comment by trishorts** [comment]
Trim has to be applied to every cell that could take a comma

**Comment by Alexander-Sol** [comment]
Changed so that the trim happens to each cell as soon as each line is read in. This should remove quotation marks inadvertently introduced into any cell/column. 

However, this will be problematic if we every introduced a field that's supposed to be enclosed in quotes.

File: EngineLayer/PsmTsv/PsmFromTsv.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ public class PsmFromTsv
 
         public PsmFromTsv(string line, char[] split, Dictionary<string, int> parsedHeader)
         {
-            var spl = line.Split(split);
+            var spl = line.Split(split).Select(p => p.Trim('\"')).ToArray();
```
Comment (avcarr2) [inline EngineLayer/PsmTsv/PsmFromTsv.cs:88]: So this is some nightmare level regex stuff, but you should be able to deal with the edge case of a field enclosed by quotes using a conditional regex: https://docs.microsoft.com/en-us/dotnet/standard/base-types/alternation-constructs-in-regular-expressions#conditional-matching-based-on-a-valid-captured-group. Might need to combine the regex with a conditional that is triggered by a pattern match.

Comment (Alexander-Sol) [reply EngineLayer/PsmTsv/PsmFromTsv.cs:88]: Good to know it's possible, but let's hope it never comes to that

File: Test/PsvTsvTest.cs [hunk 2]
```diff
@@ -25,6 +25,14 @@ public static void ReadOGlycoPsms()
             Assert.AreEqual(9, parsedPsms.Count);
         }
 
+        [Test]
+        public static void ReadExcelEditedPsms()
```
Comment (trishorts) [inline Test/PsvTsvTest.cs:29]: you could add a test for one of the fields that has quotes and show that it was parsed correctly

**Review by trishorts — COMMENTED** [review]
just a little more unit testing and we should be good.

File: Test/PsvTsvTest.cs (hunk 2 above)
Comment (Alexander-Sol) [reply Test/PsvTsvTest.cs:29]: New version of the test ensures that fragment ions (the field most often enclosed in quotes) are parsed correctly

**Review by rmmiller22 — APPROVED** [review]

**Pull request merged**